addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -73,6 +73,8 @@ class StockPicking(models.Model):
         states={'done': [('readonly', True)], 'cancel': [('readonly', True)]},
         help='Batch associated to this transfer', index=True, copy=False)
 
+    batch_user_id = fields.Many2one(related='batch_id.user_id', string="Batch Responsible", readonly=True)
```

Comment:
Related fields are readonly by default, no need to add it in explicitly 
```suggestion
    batch_user_id = fields.Many2one(related='batch_id.user_id', string="Batch Responsible")
```